test/integration/scheduler/scheduler_test.go

```diff
@@ -269,4 +273,70 @@ var _ = ginkgo.Describe("Scheduler", func() {
 			return createdJob.Spec.Suspend
 		}, framework.Timeout, framework.Interval).Should(gomega.Equal(pointer.Bool(false)))
 	})
+
+	ginkgo.It("Should schedule jobs according to the priority", func() {
+		// Initialize clusterQueue first but do not create
+		clusterQ := testing.MakeClusterQueue("cluster-queue").Resource(testing.MakeResource(corev1.ResourceCPU).
+			Flavor(testing.MakeFlavor(onDemandFlavor, "1").Obj()).Obj()).Obj()
+
+		queue := testing.MakeQueue("queue", ns.Name).ClusterQueue(clusterQ.Name).Obj()
```

Comment:
if you remove the CQ, then you could delay creating the Q until after the workloads are created.

Reply:
Thanks. It will be simpler.